### DISCUSSION TITLE
[suggestion] Organize meetings/partys

### DISCUSSION THREAD

**Discussion by MightyCreak** [element-hq/element-meta#645] [Ideas]
We were discussing with my friends about the fact that FB is a better tool than Element to organize partys.
I know Element isn't meant for that, but it could be, and that would really be a nice killer feature against the messaging part from FB.

Element already has the conversation part, what is missing is the possibility to define a description, a place, an hour, and indications for the party/meeting you wish to organize, and most importantly, the possibility to tell if you're coming or not.

I would personally use this features for very different kind of meetings, whether it is professional meetings, or birthday partys, or even in single room with friends to organize frequent partys (beer nights, RPG, movie night, ...) where anyone in the room could propose a party.

Edit: replaced "Riot" with "Element"

**Comment by ara4n** [comment]
this would be cool, and could be done as a bot with a custom widget UI.  but it's relatively far down the todo list right now.  @rxl881 this reminds me of the google calendar bot integration idea somewhat?

**Comment by Croydon** [comment]
Also nice to organise and communicate(!) project meetings.

**Comment by rsip22** [comment]
I think this could be a very nice-have.

I'm not sure if I understood you correctly, @ara4n, but as a Riot/Matrix user, may I say that in the purpose of keeping the service independent, I don't like the idea of attaching Google Calendar to this feature? I'd rather see a version which allows to use something more generic like icalendar - which _then_ may be integrated by the client of the user's chosing (whether Google, or Thunderbird, or whatever).

**Comment by rxl881** [comment]
This is something that I would really like to see (and actually started in a number of forms over Christmas, but never got round to finishing). 

There are actually a couple of vaguely related features:

* Fully calendar integrated meet-up / event planning
* Simple, consensus-driven event date planning app (like doodle.com)
* Generic option polling app (e.g. which pub should we go to)

I looked at the potential for embedding existing apps (like doodle) in custom widgets, but for various reasons (either the apps being too heavy-weight, or having restrictive licensing), I couldn't find any that were particularly suitable.

I then moved on to starting to create simple, server-backed applications for the examples mentioned above, but haven't got round to finishing them off, or getting them in to a state where they can be passed on to others yet.

However, as a result of all of the recent evolution in the v2 widget spec. (https://docs.google.com/document/d/1uPF7XWY_dXTKVKV7jZQ2KmsI19wn9-kFRgQ1tFQP7wQ/edit?usp=sharing) we will soon be able to take a different, much more light-wight  approach to this sort of feature, which could be implemented without the need for any bot, but just a client widget, with state stored in the widget data.

Please let me know if anyone has interest in implementing this type of feature / widget, if not I'll get round to it asap.

**Comment by afranke** [comment]
> Simple, consensus-driven event date planning app (like doodle.com)

Could be as simple as a [Framadate](https://framadate.org/) widget.

> Generic option polling app (e.g. which pub should we go to)

Framadate could work for that too.

**Comment by hex-m** [comment]
I don't know of any prior art (other messaging services doing this) but having "event" as a native data type would be immensely useful. While we naturally share the **where** in the form of coordinates, the **when** is still an unstructured string.

Being able to post an event in some standard format (iCalendar, JSCalendar) and make the client do the right thing (render beautifully, ask if you want to add it to your calendar, share it with other rooms/friends, show all events in a room) would be a great start.

A window that makes it easier to create and share events would make the feature more accessible.

A special type of RSVP-poll would be nice, but for my use cases reactions are probably good enough.

This could lead to having "calendar-rooms" in your space where you have one containing birthdays and another one for meetings. Then, a global calendar view over all your rooms would become handy.

Architecture-wise it could also be implemented as one Matrix-room representing one event.. :thinking:

PS: There is a lengthy discussion about this over at [Signal](https://community.signalusers.org/t/encrypted-calendar-in-signal/18395).

**Comment by Mikaela** [comment]
I don't know about doing this in Element as there is already Mobilizon which is dedicates for pages/events/groups directly challenging Facebook.

If it has to be on Matrix, [Effektio](https://effektio.org/) is a dedicated Matrix-based app for it.

**Reply by hex-m** [reply]
Thanks, this looks very promising! It's probably too early to judge how meetings/events will be implemented in Effektio. I couldn't find anything in the roadmaps. Maybe @gnunicorn has something to say about this.

**Reply by gnunicorn** [reply]
Hey @hex-m , thanks @Mikaela for the mention.

Hosting Meetups and User-Groups are actually some of the potential use cases I see for [Effektio](https://effektio.org) indeed. But this is pretty complicated, also UI-wise, and we haven't even started with the user flow there. We are still in the depth of [building the general purpose state machine to handle simple objects like Task Lists](https://github.com/effektio/effektio/pull/188) and Shared docs. But if there is wider interest and/or user groups that want that, I'm happy to talk (and might bump that use case up in the roadmap a bit more).

**Comment by aderumier** [comment]
Hi, a matrix meeting widget has been released !

https://github.com/nordeck/matrix-meetings
